gruntfile.js

```diff
@@ -170,153 +381,93 @@ module.exports = function (grunt) {
-                    tsconfig: './extensions/applicationinsights-properties-js/Tests/tsconfig.json',
-                    src: [ './extensions/applicationinsights-properties-js/Tests/Unit/src/**/*.ts' ],
-                    out: './extensions/applicationinsights-properties-js/Tests/Unit/dist/prop.tests.js'
-                },
-                reactnativetests: {
-                    tsconfig: './extensions/applicationinsights-react-native/Tests/tsconfig.json',
-                    src: [ './extensions/applicationinsights-react-native/Tests/Unit/src/**/*.ts' ],
-                    out: './extensions/applicationinsights-react-native/Tests/Unit/dist/reactnativeplugin.tests.js'
-                },
-                reacttests: {
-                    tsconfig: './extensions/applicationinsights-react/Tests/tsconfig.json',
-                    src: [ './extensions/applicationinsights-react/Tests/Unit/src/**/*.ts' ],
-                    out: './extensions/applicationinsights-react/Tests/Unit/dist/reactplugin.tests.js'
-                },
-                depsunittest: {
-                    tsconfig: './extensions/applicationinsights-dependencies-js/Tests/tsconfig.json',
-                    src: [
-                        './extensions/applicationinsights-dependencies-js/Tests/Unit/src/**/*.ts'
-                    ],
-                    out: './extensions/applicationinsights-dependencies-js/Tests/Unit/dist/dependencies.tests.js'
-                },
-                aichanneltest: {
-                    tsconfig: './channels/applicationinsights-channel-js/Tests/tsconfig.json',
-                    src: [
-                        './channels/applicationinsights-channel-js/Tests/Unit/src/**/*.ts'
-                    ],
-                    out: './channels/applicationinsights-channel-js/Tests/Unit/dist/aichannel.tests.js'
-                },
-                rollupes3test: {
-                    tsconfig: './tools/rollup-es3/Tests/tsconfig.json',
-                    src: [
-                        './tools/rollup-es3/Tests/Unit/src/**/*.ts'
-                    ],
-                    out: './tools/rollup-es3/Tests/Unit/dist/es3rolluptests.js'
-                },
-                shimstest: {
-                    tsconfig: './tools/shims/Tests/tsconfig.json',
-                    src: [
-                        './tools/shims/Tests/Unit/src**/*.ts'
-                    ],
-                    out: './tools/shims/Tests/Unit/dist/shimstests.js'
-                }
-            }
-        });
+        }));
     
         function tsBuildActions(name, addTests, replaceName) {
             var actions = [
                 "eslint-ts:" + name + "-lint-fix"
             ];
     
+            var aiMinifyConfig = theBuildConfig["ai-minify"] || {};
             var gruntTsConfig = theBuildConfig["ts"];
             if (replaceName) {
+
                 actions.push("string-replace:" + replaceName);
+                if (aiMinifyConfig[name]) {
```

Comment:
When adding the minification task, it will actually "compile" the source file twice with and without minification to check that the code and references are valid and not broken before or after